File: ApiProject/Helpers/UserParams.cs
namespace ApiProject.Helpers
{
    // class representing the user input for pagination.
    // since it inherits PaginationParams class , this class will contain 
    // access to all the parameters present in PaginationParams class. 
    public class UserParams : PaginationParams{

        public string CurrentUsername {get;set;}
        public string Gender {get;set;}

        public int MinAge {get;set;}
        public int MaxAge {get;set;}
        public string orderBy {get;set;} = "lastActive";
             
    }
}

File: WebClient/src/app/_CustomModels/userParams.ts
import { User } from "./user";

export class userParams{
    constructor(user : User){
        this.gender = user.gender === "male"?"female":"male";
    }

    minAge:number = 18;
    maxAge:number = 99;
    gender:string;
    pageNumber:number = 1;
    pageSize:number = 5;
    orderBy:string = "lastActive";

}